lib/workers/global/limits.ts

```diff
@@ -37,3 +38,159 @@ export function isLimitReached(key: Limit): boolean {
   const { max, current } = limit;
   return max - current <= 0;
 }
+
+export type CountName = 'ConcurrentPRs' | 'HourlyPRs' | 'Branches';
+
+type BranchLimitName =
+  | 'branchConcurrentLimit'
+  | 'prConcurrentLimit'
+  | 'prHourlyLimit';
+
+export const counts = new Map<CountName, number>();
+
+export function setCount(key: CountName, val: number): void {
+  const count = val;
+  counts.set(key, count);
+  logger.debug(`${key} count = ${count}`);
+}
+
+export function incCountValue(key: CountName, incBy = 1): void {
+  const count = counts.get(key) ?? 0;
+  counts.set(key, count + incBy);
+}
+
+function handleOtherLimits(
+  key: Exclude<CountName, 'HourlyPRs'>,
+  config: BranchConfig,
+): boolean {
+  const limitKey =
+    key === 'Branches' ? 'branchConcurrentLimit' : 'prConcurrentLimit';
+
+  // calculate the limits for this branch
+  const hourlyLimit = calcLimit(config.upgrades, 'prHourlyLimit');
+  const limitValue = calcLimit(config.upgrades, limitKey);
+
+  const hourlyPrCount =
+    counts.get('HourlyPRs') ??
+    // istanbul ignore next: should not happen
+    0;
```

Comment:
If this is unexpected behavior, we should print an warn log message. Same with other occurences

Reply:
It never happens as we set the count for hourlyPRs, concurrent PRs and concurrent branches before we start branch execution. But, we have to handle that case as native `Map` has return type `<T>|undefined`.

Should we still warn?

Reply:
Yes, so we are seeing this in logs and have pointer to what is happening.